### ISSUE TITLE
Add Channel name as a name variable

### ISSUE THREAD

**Issue by FranciscoTrigo** [meeb/tubesync#270]
I would like to request the addition of the channel name as a variable name.

**Comment by meeb** [comment]
Is this not covered by `{source}` or `{source_full}`?

**Comment by FranciscoTrigo** [comment]
Not when the source is a playlist, it will take the source to be whatever name I put for the source.
So for example, here: 
![image](https://user-images.githubusercontent.com/153349/193444456-fd72bdbc-a27a-479f-8840-9822a75c8783.png)
Videos will be downloaded to "NintendoVHS/Nintendo VHS/Video Title/video_title.mkv"

My use case is that I put all videos that I want to download in a playlist and use that playlist as a source. I put videos from different channels in the playlist, so it would be nice if the videos automatically downloaded in directories with their channel name.

**Comment by meeb** [comment]
OK. Can you give me an example YouTube video from your playlist and I'll see what metadata is available. It wasn't available last time I checked this, but that was over a year ago and this changes quite often so I can check it again. This wasn't available because the channel name wasn't in the per-video metadata.

**Comment by FranciscoTrigo** [comment]
https://www.youtube.com/watch?v=jHepjitYJmM

This one for example

**Comment by meeb** [comment]
Ah, it looks like it's now available in the metadata for certain videos again:

```json
    "uploader": "JonTronShow",
    "uploader_id": "JonTronShow",
```

Is that the value you would want as a `{channel}` variable?

**Comment by FranciscoTrigo** [comment]
Yes

**Comment by meeb** [comment]
OK, I'll pop it into the next release. Note this value has been added, removed, re-added and removed again over the last year or two so while it may work now it might not if remote APIs update again.

**Comment by vodkapmp** [comment]
Doesn't seem like this has been added yet, definitely a feature that I'd love to have.

**Comment by jontstaz** [comment]
@meeb - Has this been added? Would really like this as I'm migrating from standard YT-DL where I basically just added my Watch Later playlist as the 1 and only source and would just add any vids I wanted downloaded into my Watch Later. It would save the videos in `/downloads/{channel_name}/{video_info..}.mp4` which is how I wanted TubeSync to behave.

**Comment by meeb** [comment]
Last I checked the uploader wasn't reliably set. I'll have another look. I don't think this does what you want though @jontstaz, this just adds `{channel}` as something you can use in the media format for the file name. The path would still be `/downloads/video/[source slug]/[media format].[ext]` you could just use `{channel}` in the `[media format`].

**Comment by jontstaz** [comment]
> Last I checked the uploader wasn't reliably set. I'll have another look. I don't think this does what you want though @jontstaz, this just adds `{channel}` as something you can use in the media format for the file name. The path would still be `/downloads/video/[source slug]/[media format].[ext]` you could just use `{channel}` in the `[media format`].

Ahh hmm ok yeah that's not ideal then. What I want to be able to do is add a playlist and have the videos download into whatever `{uploader/channel}` folder belongs to the upload of the individual video from the playlist. Is that possible?

**Comment by vodkapmp** [comment]
> > Last I checked the uploader wasn't reliably set. I'll have another look. I don't think this does what you want though @jontstaz, this just adds `{channel}` as something you can use in the media format for the file name. The path would still be `/downloads/video/[source slug]/[media format].[ext]` you could just use `{channel}` in the `[media format`].
> 
> Ahh hmm ok yeah that's not ideal then. What I want to be able to do is add a playlist and have the videos download into whatever `{uploader/channel}` folder belongs to the upload of the individual video from the playlist. Is that possible?

This is exactly what I want to do too, and why I'm still sticking with FrenchyGhost's youtube-dl scripts.
The second it is implemented in TubeSync I'm switching over for sure.

**Comment by jontstaz** [comment]
> > Ahh hmm ok yeah that's not ideal then. What I want to be able to do is add a playlist and have the videos download into whatever `{uploader/channel}` folder belongs to the upload of the individual video from the playlist. Is that possible?
> 
> This is exactly what I want to do too, and why I'm still sticking with FrenchyGhost's youtube-dl scripts.
> The second it is implemented in TubeSync I'm switching over for sure.

Yeah I feel you 100%. I would think there's a very large proportion of TubeSync users/potential users who want this feature. It's definitely possible since we're both already doing it in youtube-dl which is what TubeSync is built on top of.

**Comment by jontstaz** [comment]
Any updates on this @meeb?

**Comment by meeb** [comment]
No not especially. Last I checked (several months back) the uploader name still wasn't reliable, but I've been busy lately and not spent any time on it. I'm happy to accept PRs for this though if someone wanted to have a go at implementing it.

**Comment by meeb** [comment]
The "uploader" variable seems stable, so I've added it back in. It should be available as `{uploader}` as of v0.13.1.
